test/readrows.ts

```diff
@@ -461,6 +462,45 @@ describe('Bigtable/ReadRows', () => {
     })();
   });
 
+  it.skip('pitfall: should not request full table scan during a retry on a transient error', () => {
+    const TRANSIENT_ERROR_SERVICE: ReadRowsServiceParameters = {
+      chunkSize: CHUNK_SIZE,
+      valueSize: VALUE_SIZE,
+      chunksPerResponse: CHUNKS_PER_RESPONSE,
+      keyFrom: STANDARD_KEY_FROM,
+      keyTo: STANDARD_KEY_TO,
+      deadlineExceededError: true,
+      debugLog,
+    };
+
+    function readRowsWithDeadline() {
+      service.setService({
+        ReadRows: ReadRowsImpl.createService(
+          TRANSIENT_ERROR_SERVICE
+        ) as ServerImplementationInterface,
+      });
+
+      table.createReadStream();
+    }
+    // Spy on the retry mechanism.
+    const retrySpy = sinon.spy(table, 'createReadStream');
+
+    try {
+      readRowsWithDeadline();
+      assert.fail('Should have thrown error');
+    } catch (err) {
+      if (err instanceof GoogleError) {
+        assert.equal(err.code, 'DEADLINE_EXCEEDED');
+      }
+
+      // Assert that the retry was attempted with an empty RowSet and limit 0
+      assert.strictEqual(retrySpy.callCount, 0);
```

Comment:
The assert fails here although `retrySpy` is counting the number of calls to `createReadStream` so we would expect this to be 1 because of `table.createReadStream();` above so this assertion statement doesn't actually check for something we want. After changing `retrySpy` to `requestSpy`, you could check `assert.strictEqual(requestSpy.callCount, 1);` to make sure exactly one request is made and therefore no retries are attempted.